### DISCUSSION TITLE
How do I disable mini.pairs for markdown files? (My autocmd isn't working)

### DISCUSSION THREAD

**Discussion by samuel-garmany** [nvim-mini/mini.nvim#805] [Q&A] [answered]
I want to disable mini.pairs for markdown files because it gets in the way of writing code blocks with triple ` . I wrote this: 
```lua
vim.api.nvim_create_autocmd("FileType", {
  pattern = { "gitcommit", "markdown" },
  callback = function()
    vim.b.mini_pairs_disable = true
  end,
})
```
But it gives the warning `Fields cannot be injected into the reference of `table<number, table<string, any>>` for `mini_pairs_disable`. To allow injection, add `[any]: any` to the definition.`

How do I fix this? Would it be easier to just unmap ` in that buffer because I like using everything else about the plugin?

**Comment by echasnovski** [comment, marked as answer]
Having LSP show a warning has nothing to do with why it is not working. Replace `mini_pairs_disable` with [`minipairs_disable`](https://github.com/echasnovski/mini.nvim/blob/cc603f4fe763741969d809ef059bf5ed457ce0e4/doc/mini-pairs.txt#L75) and it should work.

If you want to *only* disable `` ` `` in those filetypes, [override its buffer-local mapping](https://github.com/echasnovski/mini.nvim/blob/cc603f4fe763741969d809ef059bf5ed457ce0e4/doc/mini-pairs.txt#L30) so that it works as default. So with something like this:

```lua
vim.api.nvim_create_autocmd('FileType', {
  pattern = { 'gitcommit', 'markdown' },
  callback = function(event) vim.keymap.set('i', '`', '`', { buffer = event.buf }) end,
})
```

**Reply by samuel-garmany** [reply]
Replacing `mini_pairs_disable` didn't fix it, but your fix is even better. I really appreciate it!

**Reply by echasnovski** [reply]
> Replacing `mini_pairs_disable` didn't fix it...

It did fix it, as I've tested it prior to the answer. Although, the test was with `vim.b[event.buf].minipairs_disable = true`, so maybe that can be relevant in some cases.

**Reply by samuel-garmany** [reply]
![image](https://github.com/echasnovski/mini.nvim/assets/65299214/8500484c-6493-4447-8ea0-b0c9b026754e)

Must be, I was missing that originally. Thank you.
